src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java

```diff
@@ -86,16 +86,25 @@ public AccessLevel getAccessLevel(UserInfo userInfo) {
     }
 
     public Optional<List<Long>> getGrantedCaseReferencesForRestrictedRoles(CaseTypeDefinition caseTypeDefinition) {
+        if (applicationParams.getEnableAttributeBasedAccessControl()) {
+            return getGrantedCaseReferencesForRestrictedRolesRA(caseTypeDefinition);
+        } else {
+            return getGrantedCaseReferencesForRestrictedRolesNonRA();
+        }
+    }
+
+    private Optional<List<Long>> getGrantedCaseReferencesForRestrictedRolesRA(CaseTypeDefinition caseTypeDefinition) {
+        final List<Long> caseReferences =
+            roleAssignmentService
+                .getCaseReferencesForAGivenUser(userRepository.getUserId(), caseTypeDefinition)
+                .stream().map(Long::parseLong).collect(Collectors.toList());
+        return Optional.of(caseReferences);
+    }
+
+    private Optional<List<Long>> getGrantedCaseReferencesForRestrictedRolesNonRA() {
```

Comment:
getGrantedCaseReferencesForRestrictedRoles --> with RA

Reply:
👍

Reply:
This change is not part of my pr,  furthermore there is a getGrantedCaseReferencesForRestrictedRoles method already which calls either getGrantedCaseReferencesForRestrictedRolesNonRA or getGrantedCaseReferencesForRestrictedRolesRA . All of this is not part of my pr .